File: Backend/parkingapp/src/main/java/hu/hkristof/parkingapp/models/Car.java
package hu.hkristof.parkingapp.models;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

@Entity
@Table(name = "cars")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "plateNumber")
public class Car {
	
	@Id
	@NotBlank
	private String plateNumber;
	
	
	@ManyToOne(cascade = CascadeType.DETACH)
	private User owner;
	
	@OneToOne(cascade = CascadeType.PERSIST)
	private ParkingLot occupiedParkingLot;

	public ParkingLot getOccupiedParkingLot() {
		return occupiedParkingLot;
	}

	public void setOccupiedParkingLot(ParkingLot occupiedParkingLot) {
		this.occupiedParkingLot = occupiedParkingLot;
	}

	public String getPlateNumber() {
		return plateNumber;
	}

	public void setPlateNumber(String plateNumber) {
		this.plateNumber = plateNumber;
	}

	public User getOwner() {
		return owner;
	}

	public void setOwner(User owner) {
		this.owner = owner;
	}

	

}

File: mobile_app/lib/models/car.dart
import 'package:flutter/foundation.dart';
import 'package:mobile_app/models/parkingLot.dart';
import 'package:mobile_app/models/user.dart';

class Car {
  String plateNumber;
  ParkingLot occupiedParkingLot;
  User owner;

  Car({@required this.plateNumber, this.occupiedParkingLot, this.owner}) {
    if (this.owner != null) {
      if (this.owner.ownedCars != null) {
        this.owner.ownedCars.add(this);
      }
    }
  }
}
